### PR TITLE
feat: Add SORTBY array function

### PR DESCRIPTION
**Pull request by jpoz**
# PR Details

Introduces the `SORTBY` array function for dynamic in-cell sorting. 

## Description

The `SORTBY` function allows sorting a given array based on the values in one or more corresponding arrays (sort keys). Each sort key can specify an ascending or descending order.

It supports multi-key sorting and adheres to Excel's sorting rules for mixed data types: numbers precede strings, strings precede errors, and blanks come last. This provides powerful data manipulation capabilities directly within formulas.

The `SORTBY` function is in a new file `calcrange.go`, my thinking there was to move the range calculations into this file, which will reduce the overall size of calc.go making it easier to work with.

## Related Issue

NA

## Motivation and Context

The `SORTBY` function is a modern Excel dynamic array function (introduced in Excel 365) that was missing from excelize's formula calculation engine. This function is essential for users who need to sort data dynamically within formulas without manually rearranging cells.

Key benefits:
- Enables dynamic, formula-based sorting of arrays
- Supports multi-key sorting (up to 3 sort keys) with independent ascending/descending order control
- Follows Excel's standard type ordering for consistent behavior
- Complements existing array functions like `UNIQUE`, `XLOOKUP`, and `TRANSPOSE`

This implementation was organized into a new `calcrange.go` file to avoid further bloating the already large `calc.go` file (19k+ lines), making it easier to maintain and setting a pattern for future array functions like `SORT`, `FILTER`, and `SEQUENCE`.

### Test Coverage
- **`TestCalcSORTBY`**: Comprehensive integration tests covering:
  - Single key sorting (ascending/descending)
  - Multi-key sorting (2-3 keys with mixed orders)
  - Multi-column arrays and sort keys
  - Mixed data types (numbers, strings, blanks) following Excel ordering
  - 9 success cases and 7 error cases

- **`TestCompareSortbyValues`**: Unit tests for comparison logic:
  - Type ordering verification (Numbers < Strings < Errors < Blanks)
  - Case-insensitive string comparison ("Apple" == "apple")
  - Numeric comparison edge cases

- **`TestSORTBYEdgeCases`**: Edge case validation:
  - Single row arrays
  - Equal values (stable sort behavior)
  - Multi-key sorting where first key values are all equal

### Test Environment
- Go 1.18+
- All existing tests pass (`TestCalcCellValue` and others)
- No regressions introduced

### Validation Method
Tests use `TEXTJOIN` to flatten array results into comma-separated strings for easy assertion, following the pattern used by other array function tests in the codebase.

## Types of changes

- [ ] Docs change / refactoring / dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by jpoz** [comment]
@xuri updated 👍

File: calc.go [hunk 1]
```diff
@@ -19219,3 +19220,274 @@ func (fn *formulaFuncs) DISPIMG(argsList *list.List) formulaArg {
...
+		}
+
+		// All columns in this key are equal, check next key
+	}
+
+	// All keys are equal, maintain original order (stable sort)
+	return false
+}
+
+// compareSortbyValues compares two formulaArg values according to Excel sort rules:
```
Comment (xuri) [inline calc.go]: Using existing [`compareFormulaArg`](https://github.com/xuri/excelize/blob/418be6db19006f981bcf7d70666359a06ce5b0e9/calc.go#L15057-L15093) function instead of introducing new `compareSortbyValues` function.

File: calc.go [hunk 2]
```diff
@@ -19219,3 +19220,274 @@ func (fn *formulaFuncs) DISPIMG(argsList *list.List) formulaArg {
...
+	// Build result matrix
+	result := make([][]formulaArg, args.rows)
+	for i, row := range rowsWithKeys {
+		result[i] = row.rowData
+	}
+
+	return newMatrixFormulaArg(result)
+}
+
+// getSortbyArgs parses and validates the arguments for the SORTBY function.
```
Comment (xuri) [inline calc.go]: Reduce `getSortbyArgs` cyclomatic complexity less than 16.

File: calc.go [hunk 3]
```diff
@@ -19219,3 +19220,274 @@ func (fn *formulaFuncs) DISPIMG(argsList *list.List) formulaArg {
...
+		msg := fmt.Sprintf("SORTBY requires 2, 3, 5, or 7 arguments, received %d", argsLen)
+		errArg := newErrorFormulaArg(formulaErrorVALUE, msg)
+		return res, &errArg
+	}
+
+	// Parse first argument (array to sort)
+	firstArg := argsList.Front()
+	res.array = firstArg.Value.(formulaArg).ToList()
+	if len(res.array) == 0 {
+		errArg := newErrorFormulaArg(formulaErrorVALUE, "missing first argument to SORTBY")
```
Comment (xuri) [inline calc.go]: Add unit test cases to improve code coverage.

File: calc.go [hunk 4]
```diff
@@ -19219,3 +19220,274 @@ func (fn *formulaFuncs) DISPIMG(argsList *list.List) formulaArg {
...
+		return res, &errArg
+	}
+	if res.array[0].Type == ArgError {
+		return res, &res.array[0]
```
Comment (xuri) [inline calc.go]: Add unit test cases to coverage this line.

File: calc.go [hunk 5]
```diff
@@ -19219,3 +19220,274 @@ func (fn *formulaFuncs) DISPIMG(argsList *list.List) formulaArg {
...
+	currentArg := firstArg.Next()
+	keyCount := 0
+
+	for currentArg != nil && keyCount < 3 {
+		var key sortbyKey
+
+		// Parse by_array
+		key.byArray = currentArg.Value.(formulaArg).ToList()
+		if len(key.byArray) == 0 {
+			errArg := newErrorFormulaArg(formulaErrorVALUE, "missing by_array argument to SORTBY")
```
Comment (xuri) [inline calc.go:19316]: Unit tests uncoverd code.

File: calc.go [hunk 6]
```diff
@@ -19219,3 +19220,274 @@ func (fn *formulaFuncs) DISPIMG(argsList *list.List) formulaArg {
...
+			return res, &errArg
+		}
+		if key.byArray[0].Type == ArgError {
```
Comment (xuri) [inline calc.go:19319]: Unit tests uncoverd code.

File: calc.go [hunk 7]
```diff
@@ -19219,3 +19220,274 @@ func (fn *formulaFuncs) DISPIMG(argsList *list.List) formulaArg {
...
+			} else if keyCount == 1 && (argsLen == 5 || argsLen == 7) {
+				expectedSortOrder = true
+			} else if keyCount == 2 && argsLen == 7 {
+				expectedSortOrder = true
+			}
+
+			if sortOrderArg.Type == ArgError {
+				// Not a valid number
+				if expectedSortOrder {
+					// We expected a sort_order but got an error
```
Comment (xuri) [inline calc.go]: Unit tests uncoverd code.

**Review by xuri — CHANGES_REQUESTED** [review]
Thanks for your update, I've left some comments.

**Comment by xuri** [comment]
Hi @jpoz, I noticed that you implemented LET, TAKE and HSTACK formula functions in your fork, that's great! Feel free to create PR for that if you like, I'll certainly accept that.
